service/matching/matching_engine.go

```diff
@@ -1987,6 +2119,114 @@ func (e *matchingEngineImpl) pollTask(
 	return pm.PollTask(ctx, pollMetadata)
 }
 
+// isValidName checks if each character is a letter/number in the input string
+func isValidName(input string) bool {
+	for _, char := range input {
+		if !unicode.IsLetter(char) && !unicode.IsDigit(char) {
```

Comment:
Why not though? I find it rather...odd..for a user to come in and have their deployment group named to a non-alphanumerical sequence (something like: "hi!@#"). I know we give liberty to our customers when naming task-queues but I thought we should keep things simpler here while naming deploymentGroups/buildId's

This might be a more product related concern though I fear

Reply:
Yeah this is more of a product decision. But since we're not putting same restrictions anywhere else, I think we should avoid that here to be consistent: only the things that can break things should be prevented. In this example, it's very likely that build ID has chars such as "." or "-".

Reply:
I am going to go ahead and use "?" as the delimiter for our buildID. To the best of my product knowledge, users won't name their deployment group/BuildID with a question mark (but I see a world where they name things with :, -, !) so I'm taking that gamble here and going with it as a potential delimiter.

Reply:
We should not gamble on any particular delimiters. If it is important that we have a delimiter that doesn't appear in build ids or deployment names, then we can use any character we want and just escape it if it appears in the names. E.g.
```
func escapeChar(s string, c string) {
  s = replace(s, `\`, `\\`)
  s = replace(s, c, `\`+c)
}
workflowid = prefix + escapeChar(deploymentname, ":") + ":" + escapeChar(buildid, ":")
```

Reply:
I don't mind what David suggested - basically, the dilemma that I had was do we:

1. Tell the customer to avoid putting this delimiter in their inputs
2. Provide them with full freedom to input whatever they want in their inputs but in reality, the internal deployment name and buildID would be different since we would escape them

I had gone ahead with the easier option (1) but this discussion has me moving towards (2). Ill make the changes in a follow-up PR only due to time constraints. Thanks!

Reply:
In general we shouldn't impose limitations without good reason. It's a better user experience if anything they try just works.

The internal deployment name and build id are exactly what the user says they are. The escaping is only to construct an workflow id unambiguously